splink/files/chart_defs/tf_adjustment_chart.json

```diff
@@ -1,15 +1,12 @@
 {
-    "$schema": "https://vega.github.io/schema/vega-lite/v5.9.0.json",
+    "$schema": "https://vega.github.io/schema/vega-lite/v5.9.3.json",
     "title": {
         "text": "Term frequency adjusted match weights",
         "fontSize": 16,
         "subtitle": "For selected values, incl. the lowest and highest frequency",
         "anchor": "middle"
     },
     "config": {
-        "width": {
-            "signal": ""
```

Comment:
@samnlindsay this doesn't seem to do anything.  I'm trying to work out how to use your calculations for `width_expression` [here[([width_expression](https://github.com/moj-analytical-services/splink/blob/d9ae5ff78dce22f572f7aabc681bc1565c45a9e2/splink/term_frequencies.py#L304)). The only place that seems to affect the chart is by changing the 

`spec["config"]["view"]["discreteWidth"]`

but the value for that must be numeric.  Am I missing something?

<details><summary>spec for [vega lite editor](https://vega.github.io/editor/#/edited)</summary>

```
{
  "config": {
    "view": {"discreteWidth": 300},
    "params": [
      {
        "name": "gamma_sel",
        "bind": {
          "input": "select",
          "options": [1],
          "labels": ["Exact match (TF col: city)"],
          "name": "Gamma level:"
        },
        "value": 1
      }
    ]
  },
  "hconcat": [
    {
      "layer": [
        {
          "mark": {
            "type": "point",
            "filled": true,
            "size": 100,
            "stroke": "black",
            "strokeWidth": 1
          },
          "encoding": {
            "color": {
              "field": "log2_bf_tf",
              "scale": {"domain": [-2.5, 2.5], "scheme": "redyellowgreen"},
              "title": "TF adjustment weight",
              "type": "quantitative"
            },
            "tooltip": [
              {"field": "value", "title": "Value", "type": "nominal"},
              {
                "field": "log2_bf",
                "format": "+.3",
                "title": "Match weight",
                "type": "quantitative"
              },
              {
                "field": "log2_bf_tf",
                "format": "+.3",
                "title": "TF adjusted match weight",
                "type": "quantitative"
              },
              {
                "field": "log2_bf_final",
                "format": "+.3",
                "title": "Final match weight",
                "type": "quantitative"
              }
            ],
            "x": {
              "axis": {
                "labelAngle": -60,
                "labelFontSize": 16,
                "titleFontSize": 20
              },
              "field": "value",
              "sort": {"field": "log2_bf_final", "order": "ascending"},
              "title": "TF column value",
              "type": "nominal"
            },
            "y": {
              "axis": {
                "format": "+",
                "labelFontSize": 16,
                "titleFontSize": 18,
                "values": [
                  -5,
                  -4,
                  -3,
                  -2,
                  -1,
                  0,
                  1,
                  2,
                  3,
                  4,
                  5,
                  6,
                  7,
                  8,
                  9,
                  10,
                  11,
                  12,
                  13,
                  14,
                  15,
                  16,
                  17,
                  18,
                  19,
                  20
                ]
              },
              "field": "log2_bf_final",
              "title": "Match weight",
              "type": "quantitative"
            }
          }
        },
        {
          "mark": "rule",
          "encoding": {"y": {"field": "log2_bf", "type": "quantitative"}},
          "transform": [{"filter": "datum.gamma == gamma_sel"}]
        },
        {
          "mark": {"type": "rule", "opacity": 0.5, "strokeWidth": 2},
          "encoding": {
            "color": {
              "field": "log2_bf_tf",
              "legend": null,
              "scale": {"domain": [-2.5, 2.5], "scheme": "redyellowgreen"},
              "title": "TF adjustment weight",
              "type": "quantitative"
            },
            "x": {
              "field": "value",
              "sort": {"field": "log2_bf_final", "order": "ascending"},
              "title": "TF column value",
              "type": "nominal"
            },
            "y": {"field": "log2_bf_final", "type": "quantitative"},
            "y2": {"type": "quantitative"}
          },
          "transform": [{"filter": "datum.gamma == gamma_sel"}]
        }
      ],
      "data": {"name": "data"},
      "height": 400,
      "transform": [{"filter": "datum.gamma == gamma_sel"}]
    },
    {
      "mark": {
        "type": "bar",
        "fillOpacity": 0.8,
        "filled": true,
        "stroke": "black",
        "strokeWidth": 1
      },
      "data": {"name": "hist"},
      "encoding": {
        "color": {
          "field": "log2_bf_tf",
          "legend": null,
          "scale": {"domain": [-2.5, 2.5], "scheme": "redyellowgreen"},
          "title": "TF adjustment weight",
          "type": "quantitative"
        },
        "tooltip": [
          {"field": "log2_bf_desc", "title": "Match weight"},
          {"field": "count", "title": "Number of values"}
        ],
        "x": {
          "axis": {
            "domain": false,
            "gridOpacity": 0.5,
            "labelAlign": "center",
            "labelFontSize": 12,
            "labelOpacity": 0.5,
            "labelOverlap": true,
            "ticks": false,
            "title": "Count of values",
            "titleFontSize": 12,
            "titleOpacity": 0.5
          },
          "field": "count",
          "type": "quantitative"
        },
        "y": {
          "axis": null,
          "bin": {"step": 0.5},
          "field": "log2_bf_final",
          "type": "quantitative"
        }
      },
      "transform": [
        {
          "filter": {
            "or": [
              "datum.gamma == gamma_sel",
              {"field": "gamma", "valid": false}
            ]
          }
        }
      ],
      "view": {"stroke": "transparent"},
      "width": 100
    }
  ],
  "datasets": {
    "data": [
      {
        "value": "Souohnd-tn-Sea",
        "tf": 0.001122334455667789,
        "label_for_charts": "Exact match",
        "u_probability": 0.09307809682341518,
        "tf_col": "city",
        "tf_adjustment_weight": 1,
        "log2_bf": 2.8767947978615567,
        "gamma": 1,
        "log2_bf_tf": 6.373867143871083,
        "log2_bf_final": 9.25066194173264,
        "most_freq_rank": 173,
        "least_freq_rank": 0
      },
      {
        "value": "Lnodn",
        "tf": 0.001122334455667789,
        "label_for_charts": "Exact match",
        "u_probability": 0.09307809682341518,
        "tf_col": "city",
        "tf_adjustment_weight": 1,
        "log2_bf": 2.8767947978615567,
        "gamma": 1,
        "log2_bf_tf": 6.373867143871083,
        "log2_bf_final": 9.25066194173264,
        "most_freq_rank": 172,
        "least_freq_rank": 1
      },
      {
        "value": "Shefidle",
        "tf": 0.001122334455667789,
        "label_for_charts": "Exact match",
        "u_probability": 0.09307809682341518,
        "tf_col": "city",
        "tf_adjustment_weight": 1,
        "log2_bf": 2.8767947978615567,
        "gamma": 1,
        "log2_bf_tf": 6.373867143871083,
        "log2_bf_final": 9.25066194173264,
        "most_freq_rank": 171,
        "least_freq_rank": 2
      },
      {
        "value": "doLon",
        "tf": 0.001122334455667789,
        "label_for_charts": "Exact match",
        "u_probability": 0.09307809682341518,
        "tf_col": "city",
        "tf_adjustment_weight": 1,
        "log2_bf": 2.8767947978615567,
        "gamma": 1,
        "log2_bf_tf": 6.373867143871083,
        "log2_bf_final": 9.25066194173264,
        "most_freq_rank": 170,
        "least_freq_rank": 3
      },
      {
        "value": "Southapmtn",
        "tf": 0.001122334455667789,
        "label_for_charts": "Exact match",
        "u_probability": 0.09307809682341518,
        "tf_col": "city",
        "tf_adjustment_weight": 1,
        "log2_bf": 2.8767947978615567,
        "gamma": 1,
        "log2_bf_tf": 6.373867143871083,
        "log2_bf_final": 9.25066194173264,
        "most_freq_rank": 169,
        "least_freq_rank": 4
      },
      {
        "value": "Swndon",
        "tf": 0.001122334455667789,
        "label_for_charts": "Exact match",
        "u_probability": 0.09307809682341518,
        "tf_col": "city",
        "tf_adjustment_weight": 1,
        "log2_bf": 2.8767947978615567,
        "gamma": 1,
        "log2_bf_tf": 6.373867143871083,
        "log2_bf_final": 9.25066194173264,
        "most_freq_rank": 168,
        "least_freq_rank": 5
      },
      {
        "value": "Lnoon",
        "tf": 0.001122334455667789,
        "label_for_charts": "Exact match",
        "u_probability": 0.09307809682341518,
        "tf_col": "city",
        "tf_adjustment_weight": 1,
        "log2_bf": 2.8767947978615567,
        "gamma": 1,
        "log2_bf_tf": 6.373867143871083,
        "log2_bf_final": 9.25066194173264,
        "most_freq_rank": 167,
        "least_freq_rank": 6
      },
      {
        "value": "Belfts",
        "tf": 0.001122334455667789,
        "label_for_charts": "Exact match",
        "u_probability": 0.09307809682341518,
        "tf_col": "city",
        "tf_adjustment_weight": 1,
        "log2_bf": 2.8767947978615567,
        "gamma": 1,
        "log2_bf_tf": 6.373867143871083,
        "log2_bf_final": 9.25066194173264,
        "most_freq_rank": 166,
        "least_freq_rank": 7
      },
      {
        "value": "Newcaetle-upo-Tyns",
        "tf": 0.001122334455667789,
        "label_for_charts": "Exact match",
        "u_probability": 0.09307809682341518,
        "tf_col": "city",
        "tf_adjustment_weight": 1,
        "log2_bf": 2.8767947978615567,
        "gamma": 1,
        "log2_bf_tf": 6.373867143871083,
        "log2_bf_final": 9.25066194173264,
        "most_freq_rank": 165,
        "least_freq_rank": 8
      },
      {
        "value": "Newctsale-pon-Tyne",
        "tf": 0.001122334455667789,
        "label_for_charts": "Exact match",
        "u_probability": 0.09307809682341518,
        "tf_col": "city",
        "tf_adjustment_weight": 1,
        "log2_bf": 2.8767947978615567,
        "gamma": 1,
        "log2_bf_tf": 6.373867143871083,
        "log2_bf_final": 9.25066194173264,
        "most_freq_rank": 164,
        "least_freq_rank": 9
      },
      {
        "value": "Bolton",
        "tf": 0.015712682379349047,
        "label_for_charts": "Exact match",
        "u_probability": 0.09307809682341518,
        "tf_col": "city",
        "tf_adjustment_weight": 1,
        "log2_bf": 2.8767947978615567,
        "gamma": 1,
        "log2_bf_tf": 2.5665122218134786,
        "log2_bf_final": 5.443307019675036,
        "most_freq_rank": 9,
        "least_freq_rank": 164
      },
      {
        "value": "Newcastle-upon-Tyne",
        "tf": 0.016835016835016835,
        "label_for_charts": "Exact match",
        "u_probability": 0.09307809682341518,
        "tf_col": "city",
        "tf_adjustment_weight": 1,
        "log2_bf": 2.8767947978615567,
        "gamma": 1,
        "log2_bf_tf": 2.4669765482625645,
        "log2_bf_final": 5.343771346124122,
        "most_freq_rank": 8,
        "least_freq_rank": 165
      },
      {
        "value": "Brighton",
        "tf": 0.017957351290684626,
        "label_for_charts": "Exact match",
        "u_probability": 0.09307809682341518,
        "tf_col": "city",
        "tf_adjustment_weight": 1,
        "log2_bf": 2.8767947978615567,
        "gamma": 1,
        "log2_bf_tf": 2.373867143871083,
        "log2_bf_final": 5.2506619417326394,
        "most_freq_rank": 7,
        "least_freq_rank": 166
      },
      {
        "value": "Stoke-on-Trent",
        "tf": 0.028058361391694726,
        "label_for_charts": "Exact match",
        "u_probability": 0.09307809682341518,
        "tf_col": "city",
        "tf_adjustment_weight": 1,
        "log2_bf": 2.8767947978615567,
        "gamma": 1,
        "log2_bf_tf": 1.7300109540963584,
        "log2_bf_final": 4.606805751957915,
        "most_freq_rank": 6,
        "least_freq_rank": 167
      },
      {
        "value": "Southend-on-Sea",
        "tf": 0.030303030303030304,
        "label_for_charts": "Exact match",
        "u_probability": 0.09307809682341518,
        "tf_col": "city",
        "tf_adjustment_weight": 1,
        "log2_bf": 2.8767947978615567,
        "gamma": 1,
        "log2_bf_tf": 1.6189796417076145,
        "log2_bf_final": 4.495774439569171,
        "most_freq_rank": 5,
        "least_freq_rank": 168
      },
      {
        "value": "Sunderland",
        "tf": 0.031425364758698095,
        "label_for_charts": "Exact match",
        "u_probability": 0.09307809682341518,
        "tf_col": "city",
        "tf_adjustment_weight": 1,
        "log2_bf": 2.8767947978615567,
        "gamma": 1,
        "log2_bf_tf": 1.5665122218134788,
        "log2_bf_final": 4.443307019675036,
        "most_freq_rank": 4,
        "least_freq_rank": 169
      },
      {
        "value": "Sheffield",
        "tf": 0.03479236812570146,
        "label_for_charts": "Exact match",
        "u_probability": 0.09307809682341518,
        "tf_col": "city",
        "tf_adjustment_weight": 1,
        "log2_bf": 2.8767947978615567,
        "gamma": 1,
        "log2_bf_tf": 1.4196708334842079,
        "log2_bf_final": 4.2964656313457645,
        "most_freq_rank": 3,
        "least_freq_rank": 170
      },
      {
        "value": "Leeds",
        "tf": 0.037037037037037035,
        "label_for_charts": "Exact match",
        "u_probability": 0.09307809682341518,
        "tf_col": "city",
        "tf_adjustment_weight": 1,
        "log2_bf": 2.8767947978615567,
        "gamma": 1,
        "log2_bf_tf": 1.3294730245126296,
        "log2_bf_final": 4.206267822374186,
        "most_freq_rank": 2,
        "least_freq_rank": 171
      },
      {
        "value": "Birmingham",
        "tf": 0.04040404040404041,
        "label_for_charts": "Exact match",
        "u_probability": 0.09307809682341518,
        "tf_col": "city",
        "tf_adjustment_weight": 1,
        "log2_bf": 2.8767947978615567,
        "gamma": 1,
        "log2_bf_tf": 1.2039421424287706,
        "log2_bf_final": 4.080736940290327,
        "most_freq_rank": 1,
        "least_freq_rank": 172
      },
      {
        "value": "London",
        "tf": 0.2884399551066218,
        "label_for_charts": "Exact match",
        "u_probability": 0.09307809682341518,
        "tf_col": "city",
        "tf_adjustment_weight": 1,
        "log2_bf": 2.8767947978615567,
        "gamma": 1,
        "log2_bf_tf": -1.631757405322795,
        "log2_bf_final": 1.2450373925387617,
        "most_freq_rank": 0,
        "least_freq_rank": 173
      }
    ],
    "hist": [
      {
        "gamma": 1,
        "log2_bf": 2.8767947978615567,
        "count": 1,
        "log2_bf_tf": -1.631757405322795,
        "bin_start": 1,
        "bin_end": 1.5,
        "log2_bf_final": 1.25,
        "log2_bf_desc": "1.0-1.5"
      },
      {
        "gamma": 1,
        "log2_bf": 2.8767947978615567,
        "count": 5,
        "log2_bf_tf": 1.4277155727893402,
        "bin_start": 4,
        "bin_end": 4.5,
        "log2_bf_final": 4.25,
        "log2_bf_desc": "4.0-4.5"
      },
      {
        "gamma": 1,
        "log2_bf": 2.8767947978615567,
        "count": 1,
        "log2_bf_tf": 1.7300109540963584,
        "bin_start": 4.5,
        "bin_end": 5,
        "log2_bf_final": 4.75,
        "log2_bf_desc": "4.5-5.0"
      },
      {
        "gamma": 1,
        "log2_bf": 2.8767947978615567,
        "count": 3,
        "log2_bf_tf": 2.4691186379823757,
        "bin_start": 5,
        "bin_end": 5.5,
        "log2_bf_final": 5.25,
        "log2_bf_desc": "5.0-5.5"
      },
      {
        "gamma": 1,
        "log2_bf": 2.8767947978615567,
        "count": 11,
        "log2_bf_tf": 2.8644716298007653,
        "bin_start": 5.5,
        "bin_end": 6,
        "log2_bf_final": 5.75,
        "log2_bf_desc": "5.5-6.0"
      },
      {
        "gamma": 1,
        "log2_bf": 2.8767947978615567,
        "count": 8,
        "log2_bf_tf": 3.4036277977389036,
        "bin_start": 6,
        "bin_end": 6.5,
        "log2_bf_final": 6.25,
        "log2_bf_desc": "6.0-6.5"
      },
      {
        "gamma": 1,
        "log2_bf": 2.8767947978615567,
        "count": 8,
        "log2_bf_tf": 3.953301146796048,
        "bin_start": 6.5,
        "bin_end": 7,
        "log2_bf_final": 6.75,
        "log2_bf_desc": "6.5-7.0"
      },
      {
        "gamma": 1,
        "log2_bf": 2.8767947978615567,
        "count": 5,
        "log2_bf_tf": 4.373867143871083,
        "bin_start": 7,
        "bin_end": 7.5,
        "log2_bf_final": 7.25,
        "log2_bf_desc": "7.0-7.5"
      },
      {
        "gamma": 1,
        "log2_bf": 2.8767947978615567,
        "count": 5,
        "log2_bf_tf": 4.788904643149927,
        "bin_start": 7.5,
        "bin_end": 8,
        "log2_bf_final": 7.75,
        "log2_bf_desc": "7.5-8.0"
      },
      {
        "gamma": 1,
        "log2_bf": 2.8767947978615567,
        "count": 15,
        "log2_bf_tf": 5.373867143871082,
        "bin_start": 8,
        "bin_end": 8.5,
        "log2_bf_final": 8.25,
        "log2_bf_desc": "8.0-8.5"
      },
      {
        "gamma": 1,
        "log2_bf": 2.8767947978615567,
        "count": 112,
        "log2_bf_tf": 6.373867143871083,
        "bin_start": 9,
        "bin_end": 9.5,
        "log2_bf_final": 9.25,
        "log2_bf_desc": "9.0-9.5"
      }
    ]
  },
  "resolve": {"scale": {"color": "shared", "y": "shared"}},
  "spacing": 10,
  "title": {
    "text": "Term frequency adjusted match weights",
    "anchor": "middle",
    "fontSize": 16,
    "subtitle": "For selected values, incl. the lowest and highest frequency"
  },
  "$schema": "https://vega.github.io/schema/vega-lite/v5.9.3.json"
}
```

</details>


<details><summary>spec</summary>

```

```

</details>

Reply:
Yup, that's a relic from when I was trying to resolve the variable width issue when you change the dropdown selection (chatgpt sent me down a blind alley 😅). Hacked it another way so feel free to delete.

Reply:
thanks - will look!  (And enjoy your hols, no rush!)

Reply:
I think I've found a way to use step, which i think (?) is what you're after
https://github.com/moj-analytical-services/splink/pull/1315/commits/65108b1d6d506a7206c10f4d5eca5e8987d4e56b

<details><summary>spec</summary>
```
{
  "config": {
    "view": {
      "continuousWidth": 300,
      "continuousHeight": 300
    },
    "params": [
      {
        "name": "gamma_sel",
        "bind": {
          "input": "select",
          "options": [
            2,
            1
          ],
          "labels": [
            "2. Exact match (TF col: surname)",
            "1. Exact match (TF col: sinitial)"
          ],
          "name": "Gamma level:"
        },
        "value": 2
      }
    ]
  },
  "hconcat": [
    {
      "layer": [
        {
          "mark": {
            "type": "point",
            "filled": true,
            "size": 100,
            "stroke": "black",
            "strokeWidth": 1
          },
          "encoding": {
            "color": {
              "field": "log2_bf_tf",
              "scale": {
                "domain": [
                  -2.5,
                  2.5
                ],
                "scheme": "redyellowgreen"
              },
              "title": "TF adjustment weight",
              "type": "quantitative"
            },
            "tooltip": [
              {
                "field": "value",
                "title": "Value",
                "type": "nominal"
              },
              {
                "field": "log2_bf",
                "format": "+.3",
                "title": "Match weight",
                "type": "quantitative"
              },
              {
                "field": "log2_bf_tf",
                "format": "+.3",
                "title": "TF adjusted match weight",
                "type": "quantitative"
              },
              {
                "field": "log2_bf_final",
                "format": "+.3",
                "title": "Final match weight",
                "type": "quantitative"
              }
            ],
            "x": {
              "axis": {
                "labelAngle": -60,
                "labelFontSize": 16,
                "titleFontSize": 20
              },
              "field": "value",
              "sort": {
                "field": "log2_bf_final",
                "order": "ascending"
              },
              "title": "TF column value",
              "type": "nominal"
            },
            "y": {
              "axis": {
                "format": "+",
                "labelFontSize": 16,
                "titleFontSize": 18,
                "values": [
                  -5,
                  -4,
                  -3,
                  -2,
                  -1,
                  0,
                  1,
                  2,
                  3,
                  4,
                  5,
                  6,
                  7,
                  8,
                  9,
                  10,
                  11,
                  12,
                  13,
                  14,
                  15,
                  16,
                  17,
                  18,
                  19,
                  20
                ]
              },
              "field": "log2_bf_final",
              "title": "Match weight",
              "type": "quantitative"
            }
          }
        },
        {
          "mark": "rule",
          "encoding": {
            "y": {
              "field": "log2_bf",
              "type": "quantitative"
            }
          },
          "transform": [
            {
              "filter": "datum.gamma == gamma_sel"
            }
          ]
        },
        {
          "mark": {
            "type": "rule",
            "opacity": 0.5,
            "strokeWidth": 2
          },
          "encoding": {
            "color": {
              "field": "log2_bf_tf",
              "legend": null,
              "scale": {
                "domain": [
                  -2.5,
                  2.5
                ],
                "scheme": "redyellowgreen"
              },
              "title": "TF adjustment weight",
              "type": "quantitative"
            },
            "x": {
              "field": "value",
              "sort": {
                "field": "log2_bf_final",
                "order": "ascending"
              },
              "title": "TF column value",
              "type": "nominal"
            },
            "y": {
              "field": "log2_bf_final",
              "type": "quantitative"
            },
            "y2": {
              "type": "quantitative"
            }
          },
          "transform": [
            {
              "filter": "datum.gamma == gamma_sel"
            }
          ]
        }
      ],
      "data": {
        "name": "data"
      },
      "height": 400,
      "transform": [
        {
          "filter": "datum.gamma == gamma_sel"
        }
      ],
      "width": {
        "step": 20
      }
    },
    {
      "mark": {
        "type": "bar",
        "fillOpacity": 0.8,
        "filled": true,
        "stroke": "black",
        "strokeWidth": 1
      },
      "data": {
        "name": "hist"
      },
      "encoding": {
        "color": {
          "field": "log2_bf_tf",
          "legend": null,
          "scale": {
            "domain": [
              -2.5,
              2.5
            ],
            "scheme": "redyellowgreen"
          },
          "title": "TF adjustment weight",
          "type": "quantitative"
        },
        "tooltip": [
          {
            "field": "log2_bf_desc",
            "title": "Match weight"
          },
          {
            "field": "count",
            "title": "Number of values"
          }
        ],
        "x": {
          "axis": {
            "domain": false,
            "gridOpacity": 0.5,
            "labelAlign": "center",
            "labelFontSize": 12,
            "labelOpacity": 0.5,
            "labelOverlap": true,
            "ticks": false,
            "title": "Count of values",
            "titleFontSize": 12,
            "titleOpacity": 0.5
          },
          "field": "count",
          "type": "quantitative"
        },
        "y": {
          "axis": null,
          "bin": {
            "step": 0.5
          },
          "field": "log2_bf_final",
          "type": "quantitative"
        }
      },
      "transform": [
        {
          "filter": {
            "or": [
              "datum.gamma == gamma_sel",
              {
                "field": "gamma",
                "valid": false
              }
            ]
          }
        }
      ],
      "view": {
        "stroke": "transparent"
      },
      "width": 100
    }
  ],
  "datasets": {
    "data": [
      {
        "value": "0",
        "tf": 0.443,
        "label_for_charts": "1. Exact match",
        "u_probability": 0.5016603878233526,
        "tf_col": "sinitial",
        "tf_adjustment_weight": 1.0,
        "log2_bf": 0.1088751384882774,
        "gamma": 1,
        "log2_bf_tf": 0.17940432553474403,
        "log2_bf_final": 0.2882794640230214,
        "most_freq_rank": 1,
        "least_freq_rank": 0
      },
      {
        "value": "Coilns",
        "tf": 0.0010834236186348862,
        "label_for_charts": "2. Exact match",
        "u_probability": 0.0074429557488431336,
        "tf_col": "surname",
        "tf_adjustment_weight": 1.0,
        "log2_bf": 5.946524694755428,
        "gamma": 2,
        "log2_bf_tf": 2.780278211598852,
        "log2_bf_final": 8.72680290635428,
        "most_freq_rank": 320,
        "least_freq_rank": 0
      },
      {
        "value": "Wlliams",
        "tf": 0.0010834236186348862,
        "label_for_charts": "2. Exact match",
        "u_probability": 0.0074429557488431336,
        "tf_col": "surname",
        "tf_adjustment_weight": 1.0,
        "log2_bf": 5.946524694755428,
        "gamma": 2,
        "log2_bf_tf": 2.780278211598852,
        "log2_bf_final": 8.72680290635428,
        "most_freq_rank": 319,
        "least_freq_rank": 1
      },
      {
        "value": "9",
        "tf": 0.557,
        "label_for_charts": "1. Exact match",
        "u_probability": 0.5016603878233526,
        "tf_col": "sinitial",
        "tf_adjustment_weight": 1.0,
        "log2_bf": 0.1088751384882774,
        "gamma": 1,
        "log2_bf_tf": -0.1509663032502694,
        "log2_bf_final": -0.04209116476199201,
        "most_freq_rank": 0,
        "least_freq_rank": 1
      },
      {
        "value": "Crate",
        "tf": 0.0010834236186348862,
        "label_for_charts": "2. Exact match",
        "u_probability": 0.0074429557488431336,
        "tf_col": "surname",
        "tf_adjustment_weight": 1.0,
        "log2_bf": 5.946524694755428,
        "gamma": 2,
        "log2_bf_tf": 2.780278211598852,
        "log2_bf_final": 8.72680290635428,
        "most_freq_rank": 318,
        "least_freq_rank": 2
      },
      {
        "value": "William ",
        "tf": 0.0010834236186348862,
        "label_for_charts": "2. Exact match",
        "u_probability": 0.0074429557488431336,
        "tf_col": "surname",
        "tf_adjustment_weight": 1.0,
        "log2_bf": 5.946524694755428,
        "gamma": 2,
        "log2_bf_tf": 2.780278211598852,
        "log2_bf_final": 8.72680290635428,
        "most_freq_rank": 317,
        "least_freq_rank": 3
      },
      {
        "value": "Wood",
        "tf": 0.0010834236186348862,
        "label_for_charts": "2. Exact match",
        "u_probability": 0.0074429557488431336,
        "tf_col": "surname",
        "tf_adjustment_weight": 1.0,
        "log2_bf": 5.946524694755428,
        "gamma": 2,
        "log2_bf_tf": 2.780278211598852,
        "log2_bf_final": 8.72680290635428,
        "most_freq_rank": 316,
        "least_freq_rank": 4
      },
      {
        "value": "naFcis",
        "tf": 0.0010834236186348862,
        "label_for_charts": "2. Exact match",
        "u_probability": 0.0074429557488431336,
        "tf_col": "surname",
        "tf_adjustment_weight": 1.0,
        "log2_bf": 5.946524694755428,
        "gamma": 2,
        "log2_bf_tf": 2.780278211598852,
        "log2_bf_final": 8.72680290635428,
        "most_freq_rank": 315,
        "least_freq_rank": 5
      },
      {
        "value": "Stevenson",
        "tf": 0.0010834236186348862,
        "label_for_charts": "2. Exact match",
        "u_probability": 0.0074429557488431336,
        "tf_col": "surname",
        "tf_adjustment_weight": 1.0,
        "log2_bf": 5.946524694755428,
        "gamma": 2,
        "log2_bf_tf": 2.780278211598852,
        "log2_bf_final": 8.72680290635428,
        "most_freq_rank": 314,
        "least_freq_rank": 6
      },
      {
        "value": "Tamhs",
        "tf": 0.0010834236186348862,
        "label_for_charts": "2. Exact match",
        "u_probability": 0.0074429557488431336,
        "tf_col": "surname",
        "tf_adjustment_weight": 1.0,
        "log2_bf": 5.946524694755428,
        "gamma": 2,
        "log2_bf_tf": 2.780278211598852,
        "log2_bf_final": 8.72680290635428,
        "most_freq_rank": 313,
        "least_freq_rank": 7
      },
      {
        "value": "Hli",
        "tf": 0.0010834236186348862,
        "label_for_charts": "2. Exact match",
        "u_probability": 0.0074429557488431336,
        "tf_col": "surname",
        "tf_adjustment_weight": 1.0,
        "log2_bf": 5.946524694755428,
        "gamma": 2,
        "log2_bf_tf": 2.780278211598852,
        "log2_bf_final": 8.72680290635428,
        "most_freq_rank": 312,
        "least_freq_rank": 8
      },
      {
        "value": "Benjamin ",
        "tf": 0.0010834236186348862,
        "label_for_charts": "2. Exact match",
        "u_probability": 0.0074429557488431336,
        "tf_col": "surname",
        "tf_adjustment_weight": 1.0,
        "log2_bf": 5.946524694755428,
        "gamma": 2,
        "log2_bf_tf": 2.780278211598852,
        "log2_bf_final": 8.72680290635428,
        "most_freq_rank": 311,
        "least_freq_rank": 9
      },
      {
        "value": "Smith",
        "tf": 0.010834236186348862,
        "label_for_charts": "2. Exact match",
        "u_probability": 0.0074429557488431336,
        "tf_col": "surname",
        "tf_adjustment_weight": 1.0,
        "log2_bf": 5.946524694755428,
        "gamma": 2,
        "log2_bf_tf": -0.5416498832885105,
        "log2_bf_final": 5.404874811466918,
        "most_freq_rank": 9,
        "least_freq_rank": 311
      },
      {
        "value": "Hill",
        "tf": 0.010834236186348862,
        "label_for_charts": "2. Exact match",
        "u_probability": 0.0074429557488431336,
        "tf_col": "surname",
        "tf_adjustment_weight": 1.0,
        "log2_bf": 5.946524694755428,
        "gamma": 2,
        "log2_bf_tf": -0.5416498832885105,
        "log2_bf_final": 5.404874811466918,
        "most_freq_rank": 8,
        "least_freq_rank": 312
      },
      {
        "value": "Noah ",
        "tf": 0.010834236186348862,
        "label_for_charts": "2. Exact match",
        "u_probability": 0.0074429557488431336,
        "tf_col": "surname",
        "tf_adjustment_weight": 1.0,
        "log2_bf": 5.946524694755428,
        "gamma": 2,
        "log2_bf_tf": -0.5416498832885105,
        "log2_bf_final": 5.404874811466918,
        "most_freq_rank": 7,
        "least_freq_rank": 313
      },
      {
        "value": "Roberts",
        "tf": 0.011917659804983749,
        "label_for_charts": "2. Exact match",
        "u_probability": 0.0074429557488431336,
        "tf_col": "surname",
        "tf_adjustment_weight": 1.0,
        "log2_bf": 5.946524694755428,
        "gamma": 2,
        "log2_bf_tf": -0.6791534070384453,
        "log2_bf_final": 5.267371287716983,
        "most_freq_rank": 6,
        "least_freq_rank": 314
      },
      {
        "value": "Cooper",
        "tf": 0.013001083423618635,
        "label_for_charts": "2. Exact match",
        "u_probability": 0.0074429557488431336,
        "tf_col": "surname",
        "tf_adjustment_weight": 1.0,
        "log2_bf": 5.946524694755428,
        "gamma": 2,
        "log2_bf_tf": -0.8046842891223042,
        "log2_bf_final": 5.141840405633125,
        "most_freq_rank": 5,
        "least_freq_rank": 315
      },
      {
        "value": "Owen",
        "tf": 0.014084507042253521,
        "label_for_charts": "2. Exact match",
        "u_probability": 0.0074429557488431336,
        "tf_col": "surname",
        "tf_adjustment_weight": 1.0,
        "log2_bf": 5.946524694755428,
        "gamma": 2,
        "log2_bf_tf": -0.9201615065422403,
        "log2_bf_final": 5.026363188213188,
        "most_freq_rank": 4,
        "least_freq_rank": 316
      },
      {
        "value": "Thomas",
        "tf": 0.015167930660888408,
        "label_for_charts": "2. Exact match",
        "u_probability": 0.0074429557488431336,
        "tf_col": "surname",
        "tf_adjustment_weight": 1.0,
        "log2_bf": 5.946524694755428,
        "gamma": 2,
        "log2_bf_tf": -1.0270767104587524,
        "log2_bf_final": 4.919447984296676,
        "most_freq_rank": 3,
        "least_freq_rank": 317
      },
      {
        "value": "Brown",
        "tf": 0.028169014084507043,
        "label_for_charts": "2. Exact match",
        "u_probability": 0.0074429557488431336,
        "tf_col": "surname",
        "tf_adjustment_weight": 1.0,
        "log2_bf": 5.946524694755428,
        "gamma": 2,
        "log2_bf_tf": -1.9201615065422402,
        "log2_bf_final": 4.026363188213188,
        "most_freq_rank": 2,
        "least_freq_rank": 318
      },
      {
        "value": "Jones",
        "tf": 0.032502708559046585,
        "label_for_charts": "2. Exact match",
        "u_probability": 0.0074429557488431336,
        "tf_col": "surname",
        "tf_adjustment_weight": 1.0,
        "log2_bf": 5.946524694755428,
        "gamma": 2,
        "log2_bf_tf": -2.1266123840096665,
        "log2_bf_final": 3.819912310745762,
        "most_freq_rank": 1,
        "least_freq_rank": 319
      },
      {
        "value": "Taylor",
        "tf": 0.044420368364030335,
        "label_for_charts": "2. Exact match",
        "u_probability": 0.0074429557488431336,
        "tf_col": "surname",
        "tf_adjustment_weight": 1.0,
        "log2_bf": 5.946524694755428,
        "gamma": 2,
        "log2_bf_tf": -2.5772737930192315,
        "log2_bf_final": 3.369250901736197,
        "most_freq_rank": 0,
        "least_freq_rank": 320
      }
    ],
    "hist": [
      {
        "gamma": 1,
        "log2_bf": 0.1088751384882774,
        "count": 1,
        "log2_bf_tf": -0.1509663032502694,
        "bin_start": -0.5,
        "bin_end": 0.0,
        "log2_bf_final": -0.25,
        "log2_bf_desc": "-0.5-0.0"
      },
      {
        "gamma": 1,
        "log2_bf": 0.1088751384882774,
        "count": 1,
        "log2_bf_tf": 0.17940432553474403,
        "bin_start": 0.0,
        "bin_end": 0.5,
        "log2_bf_final": 0.25,
        "log2_bf_desc": "0.0-0.5"
      },
      {
        "gamma": 2,
        "log2_bf": 5.946524694755428,
        "count": 1,
        "log2_bf_tf": -2.5772737930192315,
        "bin_start": 3.0,
        "bin_end": 3.5,
        "log2_bf_final": 3.25,
        "log2_bf_desc": "3.0-3.5"
      },
      {
        "gamma": 2,
        "log2_bf": 5.946524694755428,
        "count": 1,
        "log2_bf_tf": -2.1266123840096665,
        "bin_start": 3.5,
        "bin_end": 4.0,
        "log2_bf_final": 3.75,
        "log2_bf_desc": "3.5-4.0"
      },
      {
        "gamma": 2,
        "log2_bf": 5.946524694755428,
        "count": 1,
        "log2_bf_tf": -1.9201615065422402,
        "bin_start": 4.0,
        "bin_end": 4.5,
        "log2_bf_final": 4.25,
        "log2_bf_desc": "4.0-4.5"
      },
      {
        "gamma": 2,
        "log2_bf": 5.946524694755428,
        "count": 1,
        "log2_bf_tf": -1.0270767104587524,
        "bin_start": 4.5,
        "bin_end": 5.0,
        "log2_bf_final": 4.75,
        "log2_bf_desc": "4.5-5.0"
      },
      {
        "gamma": 2,
        "log2_bf": 5.946524694755428,
        "count": 6,
        "log2_bf_tf": -0.6714914754280868,
        "bin_start": 5.0,
        "bin_end": 5.5,
        "log2_bf_final": 5.25,
        "log2_bf_desc": "5.0-5.5"
      },
      {
        "gamma": 2,
        "log2_bf": 5.946524694755428,
        "count": 24,
        "log2_bf_tf": -0.11347903147401016,
        "bin_start": 5.5,
        "bin_end": 6.0,
        "log2_bf_final": 5.75,
        "log2_bf_desc": "5.5-6.0"
      },
      {
        "gamma": 2,
        "log2_bf": 5.946524694755428,
        "count": 26,
        "log2_bf_tf": 0.36729974546133015,
        "bin_start": 6.0,
        "bin_end": 6.5,
        "log2_bf_final": 6.25,
        "log2_bf_desc": "6.0-6.5"
      },
      {
        "gamma": 2,
        "log2_bf": 5.946524694755428,
        "count": 28,
        "log2_bf_tf": 0.7802782115988516,
        "bin_start": 6.5,
        "bin_end": 7.0,
        "log2_bf_final": 6.75,
        "log2_bf_desc": "6.5-7.0"
      },
      {
        "gamma": 2,
        "log2_bf": 5.946524694755428,
        "count": 23,
        "log2_bf_tf": 1.1953157108776962,
        "bin_start": 7.0,
        "bin_end": 7.5,
        "log2_bf_final": 7.25,
        "log2_bf_desc": "7.0-7.5"
      },
      {
        "gamma": 2,
        "log2_bf": 5.946524694755428,
        "count": 37,
        "log2_bf_tf": 1.7802782115988534,
        "bin_start": 7.5,
        "bin_end": 8.0,
        "log2_bf_final": 7.75,
        "log2_bf_desc": "7.5-8.0"
      },
      {
        "gamma": 2,
        "log2_bf": 5.946524694755428,
        "count": 173,
        "log2_bf_tf": 2.7802782115988443,
        "bin_start": 8.5,
        "bin_end": 9.0,
        "log2_bf_final": 8.75,
        "log2_bf_desc": "8.5-9.0"
      }
    ]
  },
  "resolve": {
    "scale": {
      "color": "shared",
      "y": "shared"
    }
  },
  "spacing": 10,
  "title": {
    "text": "Term frequency adjusted match weights",
    "anchor": "middle",
    "fontSize": 16,
    "subtitle": "For selected values, incl. the lowest and highest frequency"
  },
  "$schema": "https://vega.github.io/schema/vega-lite/v5.9.3.json"
}
```
</details>